### PR TITLE
make prefresh optional

### PR DESCRIPTION
**Pull request by JoviDeCroock**
This seperates wmr further away from Preact

### CODE REVIEW

**Comment by developit** [comment]
I wonder if it would make sense to infer this from the existence of `pkg.dependencies.preact`. Thoughts?

**Comment by JoviDeCroock** [comment]
@developit we are not sure of that, they could have an empty pkg.json and stream in Preact

**Comment by developit** [comment]
@JoviDeCroock right, it's not a perfect 1:1, I just meant as a less direct signal of whether to include prefresh or not. We scaffold the dependency in `create-wmr` and I would assume any template project would do so as well.

As long as the polarity is set like you have it here (default to including prefresh, `--no-preact` to disable) that seems okay. Just I was thinking it would be nice if it was a bit less explicit, since we don't have any other flags people would normally have to think about passing.

**Comment by JoviDeCroock** [comment]
Yeah, what would you prefer? I think the features one is something I like the most personally. The implicit disable feels a bit hard with the npm-streaming however I would understand it

File: src/cli.js [hunk 1]
```diff
@@ -40,7 +40,9 @@ prog
 	.option('--sourcemap', 'Enable Source Maps')
 	.option('--profile', 'Generate build statistics')
 	.option('--reload', 'Switch off hmr and reload on file saves')
+	.option('--no-preact', 'Switch wmr to Preact specific mode')
```
Comment (developit) [inline src/cli.js:43]:
```suggestion
	.option('--no-preact', 'Disable Preact-specific features')
```

**Comment by developit** [comment]
I think my preference is for the `features` object as well - it means it would be easy to make a plugin that disables prefresh and changes other options (jsx pragma, etc).

File: src/wmr-middleware.js [hunk 2]
```diff
@@ -53,7 +53,8 @@ export default function wmrMiddleware({
 	aliases,
 	onError = NOOP,
 	onChange = NOOP,
-	plugins
+	plugins,
+	features = { preact: true }
```
Comment (developit) [inline src/wmr-middleware.js:57]: The defaults here should probably go in normalize-options:
https://github.com/preactjs/wmr/blob/a49b3b1505468db2af490b1ae2def93c474120f8/src/lib/normalize-options.js#L20

**Review by marvinhagemeister — APPROVED** [review]

**Pull request merged**